R/cov_struct.R

```diff
@@ -272,6 +272,11 @@ cov_struct <- function(
 #'
 #' @keywords internal
 h_reconcile_cov_struct <- function(formula = NULL, covariance = NULL) {
+  assert_multi_class(covariance, c("formula", "cov_struct"), null.ok = TRUE)
+  assert_formula(formula, null.ok = FALSE)
```

Comment:
if set to null, also need to update `h_extract_covariance_terms` to allow it to work with NULL (and there should not be cases using this function with formula as NULL)